src/registration/Register.tsx

```diff
@@ -0,0 +1,199 @@
+import { centeredSpinner } from 'src/components/icons';
+import planet from 'src/images/register-planet.svg';
+import { Ajax } from 'src/libs/ajax';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import Events from 'src/libs/events';
+import { FormLabel } from 'src/libs/forms';
+import { RegistrationLogo } from 'src/libs/logos';
+import { authStore, getTerraUser, TerraUser } from 'src/libs/state';
+import { CommunicationPreferencesCheckbox } from 'src/registration/CommunicationPreferencesCheckbox';
+import { InterestInTerraCheckbox } from 'src/registration/InterestInTerraCheckbox';
+import { LabelledTextInput } from 'src/registration/LabelledTextInput';
+import validate from 'validate.js';
+
+const constraints = ({ partOfOrganization }: { partOfOrganization: boolean }) => {
+  return {
+    givenName: { presence: { allowEmpty: false } },
+    familyName: { presence: { allowEmpty: false } },
+    email: { presence: { allowEmpty: false } },
+    institute: { presence: { allowEmpty: !partOfOrganization } },
+    department: { presence: { allowEmpty: !partOfOrganization } },
+    title: { presence: { allowEmpty: !partOfOrganization } },
+  };
+};
+
+export const Register = (): ReactNode => {
+  const user: TerraUser = getTerraUser();
+  const [busy, setBusy] = useState(false);
+  const [givenName, setGivenName] = useState(user.givenName || '');
+  const [familyName, setFamilyName] = useState(user.familyName || '');
+  const [email, setEmail] = useState(user.email || '');
+  const [partOfOrganization, setPartOfOrganization] = useState(true);
+  const [institute, setInstitute] = useState(''); // keep this key as 'institute' to be backwards compatible with existing Thurloe KVs
+  const [title, setTitle] = useState('');
+  const [department, setDepartment] = useState('');
+  const [interestInTerra, setInterestInTerra] = useState('');
+  const [marketingConsent, setMarketingConsent] = useState(true);
+
+  const register = async () => {
+    try {
+      setBusy(true);
+      const orgFields = partOfOrganization
+        ? {
+            institute,
+            department,
+            title,
+          }
+        : {};
+      await Ajax().User.profile.set({
+        firstName: givenName,
+        lastName: familyName,
+        contactEmail: email,
+        interestInTerra,
+        ...orgFields,
+      });
+      await Ajax().User.setUserAttributes({ marketingConsent });
+      authStore.update((state) => ({ ...state, registrationStatus: 'registeredWithoutTos' }));
+      await refreshTerraProfile();
+      await refreshSamUserAttributes();
+      Ajax().Metrics.captureEvent(Events.user.register);
+    } catch (error) {
+      reportError('Error registering', error);
+      setBusy(false);
+    }
+  };
+  const errors = validate(
+    { givenName, familyName, email, institute, title, department },
+    constraints({ partOfOrganization })
+  );
+
+  return (
+    <div
+      role="main"
+      style={{
```

Comment:
non-blocking:  For future consideration, style-heavy elements like this div would benefit from being broken out to a small helper component, i.e. MainRegisterPanel that just offloads the large style details from having to bloat things here in this code block.

Reply:
Another possible option is to move the style object out into a separate variable.